demo/topdown_demo_with_mmdet.py

```diff
@@ -29,18 +30,22 @@ def visualize_img(args, img_path, detector, pose_estimator, visualizer,
 
     # predict bbox
     register_mmdet_modules()
-    detect_result = inference_detector(detector, img_path)
-    pred_instance = detect_result.pred_instances.cpu().numpy()
+    det_result = inference_detector(detector, img_path)
+    pred_instance = det_result.pred_instances.cpu().numpy()
     bboxes = np.concatenate(
         (pred_instance.bboxes, pred_instance.scores[:, None]), axis=1)
     bboxes = bboxes[np.logical_and(pred_instance.labels == args.det_cat_id,
                                    pred_instance.scores > args.bbox_thr)]
-    bboxes = bboxes[nms(bboxes, args.nms_thr)][:, :4]
-
+    bboxes = bboxes[nms(bboxes, args.nms_thr), :4]
     # predict keypoints
     register_mmpose_modules()
     pose_results = inference_topdown(pose_estimator, img_path, bboxes)
-    data_samples = merge_data_samples(pose_results)
+
+    if len(pose_results) > 0:
+        print(type(pose_results))
+        data_samples = merge_data_samples(pose_results)
+    else:
+        data_samples = PoseDataSample()
```

Comment:
How about handling empty `pose_results` in `merge_data_samples`? As this function might be called somewhere else. An empty input check should always be added where this function is called.